edx/edx-enterprise
# ENT-454 Refactor session management logic.

**Description:** The first commit in this PR is a refactor which moves edx-enterprise decorators out of the enterprise.utils module and into an enterprise.decorators module. The second commit in this PR is a refactor of the work done in https://github.com/edx/edx-enterprise/pull/130. I have moved the session termination logic into its own decorator so that it can be applied to only the views that need it.

**JIRA:** https://openedx.atlassian.net/browse/ENT-454

**Dependencies:** None

**Merge deadline:** ASAP

**Installation instructions:** Standard Installation with this version of edx-enterprise.

**Testing instructions:**

1. Go to https://business.sandbox.edx.org/enterprise/2b5a2956-7746-4fc9-9587-bc887ba45973/course/course-v1:edX+DemoX+Demo_Course/enroll/ and register a new user, going through the normal flow until you get to the dashboard. You should be able to submit DSC consent after the course enrollment landing page without error. Do not log out.

3. Sign out of the user on the IdP or unlinking the first user from the IdP account, and revisit the page. It should redirect to the first user's account.

4. Change the Provider Configuration to enable drop existing session.

5. Revisit the page (if you have not logged out of the first account, otherwise repeat step 2 first). It should direct you to the account creation page as if you were a new user coming in. You should then be able to go through the normal flow as expected.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
